### ISSUE TITLE
TensorFlow-GPU random GPU crashes while training

### ISSUE THREAD

**Issue by NightCrawler96** [tensorflow/tensorflow#18316]
### System information
- Custom code - Simple q learning model using Keras with TF and Open AI Gym
- Windows 10
- Tensorflow installed with pip
- TensorFlow-gpu 1.7.0
- Python 3.6.4
- CUDA 9.0, cuDNN 7
- Nvidia GeForce GTX 1050 Ti GDDR5 4GB

### Describtion
When I'm running a script randomly two types of errors happen:

-  Blue Screen with Driver Power State Failure
- TensorFlow cannot locate gpu " failed call to cuInit: CUDA_ERROR_NO_DEVICE"

This happens approximately once per 10 runs. GeForce is my secondary gpu card, so I assume, that it is not related to watchdog timer.

Result od deviceQuery:
> Device 0: "GeForce GTX 1050 Ti"
>   CUDA Driver Version / Runtime Version          9.0 / 9.0
>   CUDA Capability Major/Minor version number:    6.1
>   Total amount of global memory:                 4096 MBytes (4294967296 bytes)
>   ( 6) Multiprocessors, (128) CUDA Cores/MP:     768 CUDA Cores
>   GPU Max Clock rate:                            1620 MHz (1.62 GHz)
>   Memory Clock rate:                             3504 Mhz
>   Memory Bus Width:                              128-bit
>   L2 Cache Size:                                 1048576 bytes
>   Maximum Texture Dimension Size (x,y,z)         1D=(131072), 2D=(131072, 65536), 3D=(16384, 16384, 16384)
>   Maximum Layered 1D Texture Size, (num) layers  1D=(32768), 2048 layers
>   Maximum Layered 2D Texture Size, (num) layers  2D=(32768, 32768), 2048 layers
>   Total amount of constant memory:               65536 bytes
>   Total amount of shared memory per block:       49152 bytes
>   Total number of registers available per block: 65536
>   Warp size:                                     32
>   Maximum number of threads per multiprocessor:  2048
>   Maximum number of threads per block:           1024
>   Max dimension size of a thread block (x,y,z): (1024, 1024, 64)
>   Max dimension size of a grid size    (x,y,z): (2147483647, 65535, 65535)
>   Maximum memory pitch:                          2147483647 bytes
>   Texture alignment:                             512 bytes
>   Concurrent copy and kernel execution:          Yes with 2 copy engine(s)
>   Run time limit on kernels:                     Yes
>   Integrated GPU sharing Host Memory:            No
>   Support host page-locked memory mapping:       Yes
>   Alignment requirement for Surfaces:            Yes
>   Device has ECC support:                        Disabled
>   CUDA Device Driver Mode (TCC or WDDM):         WDDM (Windows Display Driver Model)
>   Device supports Unified Addressing (UVA):      Yes
>   Supports Cooperative Kernel Launch:            No
>   Supports MultiDevice Co-op Kernel Launch:      No
>   Device PCI Domain ID / Bus ID / location ID:   0 / 1 / 0
>   Compute Mode:
>      < Default (multiple host threads can use ::cudaSetDevice() with device simultaneously) >
> 
> deviceQuery, CUDA Driver = CUDART, CUDA Driver Version = 9.0, CUDA Runtime Version = 9.0, NumDevs = 1
> Result = PASS

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by NightCrawler96** [comment]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:  Windows 10
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.7.0
- **Python version**: 3.6.4
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: 9.0/7.0.3.11
- **GPU model and memory**: GeForce GTX 1050 Ti GDDR5 4GB
- **Exact command to reproduce**: N/A

I can add, that that BSOD happens at the end of the script, so it might be an issue with delating objects or freeing GPU memory.

**Comment by bignamehyp** [comment]
Do you have a reproducible code that contains tensorflow script only? It might be a Open AI Gym issue

**Comment by NightCrawler96** [comment]
This happened previously without Open AI Gym when I was using Jupyter notebook from Intel's course on tensorflow. 
I tried to reproduce this bug using both example code and my script but since I've updated tf-gpu to 1.8.0 issue seems to go away. I'll update if problem returns.

**Comment by NightCrawler96** [comment]
20 mins after posting previous comment error returned. When I was training my DDPG agent error occured after ~600 episodes:

> 2018-05-03 15:24:23.086336: E T:\src\github\tensorflow\tensorflow\stream_executor\cuda\cuda_event.cc:49] Error polling for event status: failed to query event: CUDA_ERROR_UNKNOWN
2018-05-03 15:24:23.091324: F T:\src\github\tensorflow\tensorflow\core\common_runtime\gpu\gpu_event_mgr.cc:208] Unexpected Event status: 1

Maybe it is related to GPU running at full speed for a long time?

I will try to reproduce this using code without open ai gym, but I have to find an example that uses gpu for extended time first.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 23 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 37 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 29 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by BBarbosa** [comment]
Hi @NightCrawler96!
I'm facing the same issue as you described. 
Here is my setup: 
- Custom code - Keras with TF on
- Windows 10
- Tensorflow installed with pip
- TensorFlow-gpu 1.8.0
- Python 3.6.2
- CUDA 9.0, cuDNN 7
- Nvidia GeForce GTX 1070 GDDR5 8GB

By the way, did you find any solution for it?

**Comment by tensorflowbutler** [comment]
It has been 15 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 30 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by bignamehyp** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by NightCrawler96** [comment]
Sorry for long time without an activity. I had some health issues and had to focus on that first.
***
I think this issue has been **solved**.
My laptop finally broke and I had to reinstall everything. With my current setup everything runned for about 9 hours using GPU, so I think it is unlikely to return again.
When I was writing this comment I figured out, that I had forgotten to install cuDNN. It was running smoothly without it.

Current setup:

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 1.9.0
- Python version: 3.6.6
- CUDA/cuDNN version: 9.2.148/None
- GPU model and memory: GeForce GTX 1050 Ti GDDR5 4GB
